client/rest-high-level/src/main/java/org/elasticsearch/client/ml/datafeed/DatafeedStats.java

```diff
@@ -0,0 +1,136 @@
+    } );
+
+    static {
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), DatafeedConfig.ID);
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), DatafeedState.STATE);
+        PARSER.declareObject(ConstructingObjectParser.optionalConstructorArg(), NodeAttributes.PARSER, NODE);
+        PARSER.declareString(ConstructingObjectParser.optionalConstructorArg(), ASSIGNMENT_EXPLANATION);
+    }
+
+    public DatafeedStats(String datafeedId, DatafeedState datafeedState, @Nullable NodeAttributes node,
+                         @Nullable String assignmentExplanation) {
+        this.datafeedId = Objects.requireNonNull(datafeedId);
+        this.datafeedState = Objects.requireNonNull(datafeedState);
+        this.node = node;
+        this.assignmentExplanation = assignmentExplanation;
+    }
+
+    public String getDatafeedId() {
+        return datafeedId;
+    }
+
+    public DatafeedState getDatafeedState() {
+        return datafeedState;
+    }
+
+    public NodeAttributes getNode() {
+        return node;
+    }
+
+    public String getAssignmentExplanation() {
+        return assignmentExplanation;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params params) throws IOException {
+        builder.startObject();
+        builder.field(DatafeedConfig.ID.getPreferredName(), datafeedId);
+        builder.field(DatafeedState.STATE.getPreferredName(), datafeedState.toString());
+        if (node != null) {
+            builder.startObject("node");
+            builder.field("id", node.getId());
+            builder.field("name", node.getName());
+            builder.field("ephemeral_id", node.getEphemeralId());
+            builder.field("transport_address", node.getTransportAddress());
+
+            builder.startObject("attributes");
+            for (Map.Entry<String, String> entry : node.getAttributes().entrySet()) {
+                if (entry.getKey().startsWith("ml.")) {
+                    builder.field(entry.getKey(), entry.getValue());
+                }
+            }
+            builder.endObject();
+            builder.endObject();
+        }
+        if (assignmentExplanation != null) {
+            builder.field(ASSIGNMENT_EXPLANATION.getPreferredName(), assignmentExplanation);
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(datafeedId, datafeedState.toString(), node, assignmentExplanation);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == null) {
+            return false;
+        }
+        if (getClass() != obj.getClass()) {
+            return false;
+        }
+        DatafeedStats other = (DatafeedStats) obj;
+        return Objects.equals(datafeedId, other.datafeedId) &&
+            Objects.equals(this.datafeedState.toString(), other.datafeedState.toString()) &&
```

Comment:
nit: there's no need to stringify datafeedState here and in `hashCode()`... or am I missing something